src/codegen/mod.rs

```diff
@@ -2492,13 +2502,22 @@ impl<'a> EnumBuilder<'a> {
                 }
             }
 
-            EnumVariation::Consts => EnumBuilder::Consts {
-                variants: vec![quote! {
-                    #( #attrs )*
-                    pub type #ident = #repr;
-                }],
-                codegen_depth: enum_codegen_depth,
-            },
+            EnumVariation::Consts => {
+                let mut variants = Vec::new();
+
+                if is_ty_named {
+                    variants.push(quote! {
+                        #( #attrs )*
+                        pub type #ident = #repr;
+                    });
+                }
+
+                EnumBuilder::Consts {
+                    repr: repr,
```

Comment:
nit: use the shorthand notation? `repr, variants, `

Reply:
Well, actually it's little effort, and I want to add another test too, so I can do this after merging.

Thank you so much for the patch!